core/src/main/java/tachyon/worker/hierarchy/StorageDir.java

```diff
@@ -273,7 +274,17 @@ public boolean copyBlock(long blockId, StorageDir dstDir) throws IOException {
       CommonUtils.cleanDirectBuffer(buffer);
     }
     if (copySuccess) {
-      dstDir.addBlockId(blockId, size, mLastBlockAccessTimeMs.get(blockId), true);
+      Long accessTime = mLastBlockAccessTimeMs.get(blockId);
```

Comment:
Would checking mToRemoveBlockIdSet.contains(blockId) work here? I think its a bit better since its not immediately obvious null accessTime means the block is scheduled for deletion.